rust_src/src/atimer.rs

```diff
@@ -0,0 +1,281 @@
+use std::os::raw::c_void;
+use std::ptr::null_mut;
+use std::mem::uninitialized;
+
+use libc::sigset_t;
+use libc::c_int;
+use libc::sigemptyset;
+use libc::sigaddset;
+use libc::pthread_sigmask;
+use libc::timespec;
+use libc::{SIGALRM, SIGINT, SIG_BLOCK, SIG_SETMASK};
+
+extern "C" {
+    static mut free_atimers: *mut Atimer;
+    static mut stopped_atimers: *mut Atimer;
+    static mut atimers: *mut Atimer;
+
+    static mut pending_signals: i32;
+}
+
+// Types of timers.
+// FIXME How large this enum is in C?
```

Comment:
Maybe  in C you can change the enum to `#define`s and typedef the enum name to a fixed size type.
**Example:**
```c
enum ATIMER_TYPE {
   ATIMER_ABSOLUTE,
   ...
}
```

Changes to:
```c
typedef uint32_t ATIMER_TYPE;

#define ATIMER_ABSOLUTE 0
...
```